build.gradle

```diff
@@ -18,6 +18,7 @@ dependencies {
     compile 'org.jetbrains:annotations:13.0'
     testCompile 'junit:junit:4.12'
     testCompile 'org.mock-server:mockserver-netty:5.2.2'
+    testCompile 'org.mockito:mockito-core:2.+'
```

Comment:
Mockito is a mocking framework for unit testing in java. It is quite popular among the community. 

In this case, It is added for testing whether the  uploadFinished method is called or not by the TusUploader.
I'm creating a mock of TusClient and then I check how many times the uploadFinished is called by TusUploader. 

As an example....

```java
 TusClient client = mock(TusClient.class);
 TusUpload upload = new TusUpload();
  ...
 // Asserts that uploadFinished is called exactly 1 time after the test execution receiving upload as parameter
 verify(client).uploadFinished(upload); 

```
It you are not confortable with adding a new testing dependency, the alternative is to extend the TusClient and write the corresponding uploadFinished implementation for each test case.

Reply:
Thank for very much for the detailed explanation. I am fine with using Mockito, especially since it's a popular tool. :+1: